Src/GoogleApis.Auth.DotNet4/OAuth2/GoogleCredential.cs

```diff
@@ -120,107 +130,52 @@ public virtual GoogleCredential CreateScoped(IEnumerable<string> scopes)
 
         #region Factory methods
 
         /// <summary>Creates a <c>GoogleCredential</c> wrapping a <see cref="ComputeCredential"/>.</summary>
         internal static GoogleCredential FromCredential(ComputeCredential credential)
         {
-            return new ComputeGoogleCredential(credential);
+            return new GoogleCredential(credential, credential, credential);
         }
 
         /// <summary>Creates a <c>GoogleCredential</c> wrapping a <see cref="ServiceAccountCredential"/>.</summary>
-        internal static GoogleCredential FromCredential(ServiceAccountCredential credential)
+        internal static GoogleCredential FromCredential(JwtServiceAccountCredential credential)
         {
-            return new ServiceAccountGoogleCredential(credential);
+            return new JwtServiceAccountGoogleCredential(credential);
         }
 
         /// <summary>Creates a <c>GoogleCredential</c> wrapping a <see cref="UserCredential"/>.</summary>
         internal static GoogleCredential FromCredential(UserCredential credential)
         {
-            return new UserGoogleCredential(credential);
+            return new GoogleCredential(credential, credential, credential);
         }
 
         #endregion
 
-        // TODO(jtattermush): Look into adjusting the API of ServiceAccountCredential, ComputeCredential
-        // and UserCredential so that they implement ITokenAccess. Then the boilerplate below will go away.
-
-        /// <summary>Wraps <c>ComputeCredential</c> as <c>GoogleCredential</c>.</summary>
-        internal class ComputeGoogleCredential : GoogleCredential
-        {
-            private readonly ComputeCredential credential;
-
-            public ComputeGoogleCredential(ComputeCredential credential)
-            {
-                this.credential = credential;
-            }
-
-            #region GoogleCredential overrides
-
-            internal override object UnderlyingCredential
-            {
-                get { return credential; }
-            }
-
-            protected override void Initialize(ConfigurableHttpClient httpClient)
-            {
-                credential.Initialize(httpClient);
-            }
-
-            protected override TokenResponse Token
-            {
-                get { return credential.Token; }
-            }
-
-            protected override Task<bool> RequestAccessTokenAsync(CancellationToken taskCancellationToken)
-            {
-                return credential.RequestAccessTokenAsync(taskCancellationToken);
-            }
-
-            #endregion
-        }
-
-        /// <summary>Wraps <c>ServiceAccountCredential</c> as <c>GoogleCredential</c>.</summary>
-        internal class ServiceAccountGoogleCredential : GoogleCredential
+        /// <summary>Wraps <c>JwtServiceAccountCredential</c> as <c>GoogleCredential</c>.</summary>
+        internal class JwtServiceAccountGoogleCredential : GoogleCredential
```

Comment:
Why do we need this class, while we deleted the others?
Let's chat. Feel free to put something in my cal before I'm leaving...